sql/core/src/test/scala/org/apache/spark/sql/DataFrameSetOperationsSuite.scala

```diff
@@ -922,12 +922,67 @@ class DataFrameSetOperationsSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-35756: unionByName support struct having same col names but different sequence") {
+    // struct having same col names but different sequence
+    var df1 = Seq((1, Struct1(1, 2))).toDF("a", "b")
+    var df2 = Seq((1, Struct2(1, 2))).toDF("a", "b")
+    var unionDF = df1.unionByName(df2)
+    var expected = Row(1, Row(1, 2)) :: Row(1, Row(2, 1)) :: Nil
+    val schema = "`a` INT,`b` STRUCT<`c1`: INT, `c2`: INT>"
+    assert(unionDF.schema.toDDL === schema)
```

Comment:
It's a bit fragile to compare the DDL string, can we compare `StructType` instance directly?

Reply:
added the StructType  comparison